tests/unit/jobserver/models/test_job_request.py

```diff
@@ -378,3 +384,25 @@ def test_jobrequestmanager_with_started_at():
     JobRequestFactory(workspace=workspace)
 
     assert JobRequest.objects.with_started_at().filter(workspace=workspace).count() == 2
+
+
+def test_jobrequest_database_name_with_no_project_number():
+    job_request = _make_job_request(project_number=None)
+    assert job_request.database_name == "default"
+
+
+def test_jobrequest_database_name_without_t1oo_permission():
+    job_request = _make_job_request(project_number=1)
+    assert job_request.database_name == "default"
+
+
+def test_jobrequest_database_name_with_t1oo_permission():
+    # This is one of the project numbers hardcoded into `t1oo_permissions.py`
+    job_request = _make_job_request(project_number=2)
+    assert job_request.database_name == "include_t1oo"
+
+
+def _make_job_request(project_number):
+    project = ProjectFactory(number=project_number)
+    workspace = WorkspaceFactory(project=project)
+    return JobRequestFactory(workspace=workspace)
```

Comment:
Could you wrap this function in a fixture?  That's how we expose combinations of factories (as per our [testing docs](https://github.com/opensafely-core/job-server/blob/main/TESTING.md#tooling)) and how we expose these builder functions as well.  There's some prior art in there, for example `build_release()`.

Reply:
Thanks. It would need to be fixture which returns a function though, wouldn't it? Otherwise I can't pass the `project_number` in. Or am I missing something?

Reply:
Yep!  That's what `build_release()` does too :)

Reply:
OK, fine. It just looked like a pointless extra level of indirection to me:
```py
def test_jobrequest_database_name_without_t1oo_permission(build_job_request):
    job_request = build_job_request(project_number=1)
    assert job_request.database_name == "default"


@pytest.fixture
def build_job_request():
    def _build_job_request(project_number):
        project = ProjectFactory(number=project_number)
        workspace = WorkspaceFactory(project=project)
        return JobRequestFactory(workspace=workspace)
    return _build_job_request
```

vs

```py
def test_jobrequest_database_name_without_t1oo_permission():
    job_request = build_job_request(project_number=1)
    assert job_request.database_name == "default"

def build_job_request(project_number):
    project = ProjectFactory(number=project_number)
    workspace = WorkspaceFactory(project=project)
    return JobRequestFactory(workspace=workspace)
```

But if it's more idiomatic I'm very happy to keep things consistent with what's done elsewhere.